acceptance/tests/cloud/remote_dev_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cloud
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/environment"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+
+	hcpgnm "github.com/hashicorp/hcp-sdk-go/clients/cloud-global-network-manager-service/preview/2022-02-15/client/global_network_manager_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-global-network-manager-service/preview/2022-02-15/models"
+	hcpcfg "github.com/hashicorp/hcp-sdk-go/config"
+	"github.com/hashicorp/hcp-sdk-go/httpclient"
+	"github.com/hashicorp/hcp-sdk-go/resource"
+)
+
+type DevTokenResponse struct {
+	Token string `json:"token"`
+}
+
+type hcp struct {
+	ResourceID   string
+	ClientID     string
+	ClientSecret string
+	AuthURL      string
+	APIHostname  string
+	ScadaAddress string
+}
+
+func TestRemoteDevCloud(t *testing.T) {
+	_, rIDok := os.LookupEnv("HCP_RESOURCE_ID")
+	_, cIDok := os.LookupEnv("HCP_CLIENT_ID")
+	_, cSECok := os.LookupEnv("HCP_CLIENT_SECRET")
+
+	if !rIDok || !cIDok || !cSECok {
+		t.Log("Must set HCP_RESOURCE_ID, HCP_CLIENT_ID and HCP_CLIENT_SECRET")
+		t.FailNow()
+	}
+
+	ctx := suite.Environment().DefaultContext(t)
+
+	kubectlOptions := ctx.KubectlOptions(t)
+	ns := kubectlOptions.Namespace
+	k8sClient := environment.KubernetesClientFromOptions(t, kubectlOptions)
+
+	var (
+		resourceSecretName     = "resource-sec-name"
+		resourceSecretKey      = "resource-sec-key"
+		resourceSecretKeyValue = os.Getenv("HCP_RESOURCE_ID")
+
+		clientIDSecretName     = "clientid-sec-name"
+		clientIDSecretKey      = "clientid-sec-key"
+		clientIDSecretKeyValue = os.Getenv("HCP_CLIENT_ID")
+
+		clientSecretName     = "client-sec-name"
+		clientSecretKey      = "client-sec-key"
+		clientSecretKeyValue = os.Getenv("HCP_CLIENT_SECRET")
+
+		apiHostSecretName = "apihost-sec-name"
+		apiHostSecretKey  = "apihost-sec-key"
+		// helloworldsvc.test.svc.cluster.local:9111
+		apiHostSecretKeyValue = "https://api.hcp.dev" //TODO this will be the name of the test service
```

Comment:
Could we set these 3 test service names as environment variables so that they can be kept secret from the internet?  We'd put them in vault and then read them in on test run.

Reply:
We can definitely do that but it's not that much of a secret as we have it in multiple repos now. I'll add them as env variables regardless.